# CIAA-CIAA
In this notebook I work with the CIAA-CIAA dataset as created by my very own crawler from [1]. Step by step I try to proof the statistics given by [2].

###### written by Nils Dyck, 15.11.2022

## Import Data
At first I import the needed libraries and the data I work with.

In [2]:
import pandas as pd

In [6]:
data = pd.read_csv('../data/cleaned/CIAA-CIAA.csv', sep = ';')
default = pd.read_csv('../data/cleaned/gruber-holzer-rauch_dataset.csv', sep = ';')
#to show the structure of the data
data.head()

Author                                              Title  \
0              Yuan Gao                              In Memoriam Sheng Yu.   
1           Kai Salomaa                              In Memoriam Sheng Yu.   
2  Janusz A. Brzozowski       In Search of Most Complex Regular Languages.   
3   Andrzej Ehrenfeucht  A Formal Framework for Processes Inspired by t...   
4    Grzegorz Rozenberg  A Formal Framework for Processes Inspired by t...   

            Co-Authors Pagination  Published      Venue  
0          Kai Salomaa        1-4       2012  CIAA 2012  
1             Yuan Gao        1-4       2012  CIAA 2012  
2                  NaN       5-24       2012  CIAA 2012  
3   Grzegorz Rozenberg      25-27       2012  CIAA 2012  
4  Andrzej Ehrenfeucht      25-27       2012  CIAA 2012

## Total Numbers
I extract the total number of papers and authors.

In [7]:
all_papers = data[['Title', 'Published']].drop_duplicates()
total_papers = len(all_papers.index)
papers_2022 = len(all_papers[all_papers.Published == 2022].index)
print(total_papers)
print(papers_2022)

707
19


As you can see there is a total number of 707 papers while 19 where published at CIAA 2022. So the total number of papers given by [2] is correct.

In [11]:
all_authors = data[['Author', 'Published']].drop_duplicates()
total_authors = len(all_authors.index)
authors_2022 = len(all_authors[all_authors.Published == 2022].index)
print(total_authors)
#authors_2022
default_authors = default[['Author', 'Year']].drop_duplicates()
print(len(default_authors.index))

1498
1454


## Sources
[1] https://dblp.uni-trier.de/db/conf/wia/index.html (15.11.2022)

[2] Gruber, H., Holzer, M., Rauch, C. (2022). On 25 Years of CIAA Through the Lens of Data Science. In: Caron, P., Mignot, L. (eds) Implementation and Application of Automata. CIAA 2022. Lecture Notes in Computer Science, vol 13266. Springer, Cham. https://doi.org/10.1007/978-3-031-07469-1_1